Source/WebKit/Configurations/_WebKit_SwiftUI.xcconfig

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2024 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "BaseTarget.xcconfig"
+
+// InstallAPI is transitively enabled through including BaseTarget.xcconfig.
+
+PRODUCT_NAME = _WebKit_SwiftUI; // Cross-import overlay projects must have this exact naming scheme.
+PRODUCT_BUNDLE_IDENTIFIER = com.apple.WebKit.$(PRODUCT_NAME:rfc1034identifier);
+INFOPLIST_FILE = Info.plist;
+
+INSTALL_PATH = $(WEBKIT_FRAMEWORKS_DIR);
+DYLIB_INSTALL_NAME_BASE = $(NORMAL_WEBKIT_FRAMEWORKS_DIR);
+
+ALTERNATE_ROOT_PATH = $(NORMAL_WEBKIT_FRAMEWORKS_DIR);
+
+OUTPUT_ALTERNATE_ROOT_PATH = $(OUTPUT_ALTERNATE_ROOT_PATH_$(USE_SYSTEM_CONTENT_PATH)_$(WK_NOT_$(WK_USE_OVERRIDE_FRAMEWORKS_DIR)));
+OUTPUT_ALTERNATE_ROOT_PATH_YES_YES = $(DSTROOT)$(ALTERNATE_ROOT_PATH)/$(FULL_PRODUCT_NAME);
+
+BUILD_LIBRARY_FOR_DISTRIBUTION = YES; // This is required for all public frameworks in the OS.
+
+SWIFT_LIBRARY_LEVEL = api;
+
+// Prevent Objective-C from importing the cross-import overlay
+SWIFT_INSTALL_OBJC_HEADER = NO;
+DEFINES_MODULE = NO;
+
+OTHER_SWIFT_FLAGS = -Xfrontend -experimental-spi-only-imports $(OTHER_SWIFT_FLAGS_AVAILABILITY_$(USE_INTERNAL_SDK));
+
+// Disable optimization for debug builds.
+SWIFT_OPTIMIZATION_LEVEL = -O;
+SWIFT_OPTIMIZATION_LEVEL[config=Debug] = -Onone;
```

Comment:
I know this is copied from WebKitSwift, but do you know if it's actually needed? It seems weird, like, of course debug builds should be `-Onone` and I would expect the build system to do that by default.

Reply:
it is needed afaik; Swift Previews complain with an error when this is omitted

Reply:
What we should really do is sink this into the proper `Debug.xcconfig` or whatever files. When I did the overlay work I wanted to keep anything even tangentially Swift-related out of the standard configs (because we weren’t sure if it’d have unwanted effects in B&I), but these days we should just do this normally.

>  I would expect the build system to do that by default.

It’s a reasonable thing to want, but IIRC all options have one singular default (even in C). Xcode doesn’t actually understand “configs”, they’re just named collections of overrides without semantics, so it has no real understand of what “Debug” is. We have to bring that all ourselves.

Reply:
I checked XCBuild and the default is indeed `-O`. Weird.